crates/bevy_reflect/bevy_reflect_derive/src/field_attributes.rs

```diff
@@ -69,85 +70,158 @@ pub(crate) enum DefaultBehavior {
+                }
+            }
+            Meta::NameValue(pair) => {
+                let path = &pair.path;
+                Err(syn::Error::new(
+                    path.span(),
+                    format!("unknown attribute parameter: {}", path.to_token_stream()),
+                ))
+            }
+            Meta::List(list) if !list.path.is_ident(REFLECT_ATTRIBUTE_NAME) => {
+                Err(syn::Error::new(list.path.span(), "unexpected property"))
+            }
+            Meta::List(list) => {
+                for nested in &list.nested {
+                    if let NestedMeta::Meta(meta) = nested {
+                        self.parse_meta(args, meta)?;
+                    }
+                }
+                Ok(())
+            }
         }
-        Meta::List(list) if !list.path.is_ident(REFLECT_ATTRIBUTE_NAME) => {
-            Err(syn::Error::new(list.path.span(), "unexpected property"))
+    }
+
+    /// Verifies `#[reflect(ignore)]` attributes are always last in the type definition.
+    fn check_ignore_order(&self, args: &ReflectFieldAttr, errors: &mut Option<syn::Error>) {
+        if args.ignore.is_active() {
+            if let Some(span) = self.last_ignored {
+                if self.is_variant {
+                    Self::combine_error(syn::Error::new(span, format!("fields marked with `#[reflect({IGNORE_ALL_ATTR})]` must come last in variant definition")), errors);
+                } else {
+                    Self::combine_error(syn::Error::new(span, format!("fields marked with `#[reflect({IGNORE_ALL_ATTR})]` must come last in type definition")), errors);
+                }
+            }
         }
-        Meta::List(list) => {
-            for nested in &list.nested {
-                if let NestedMeta::Meta(meta) = nested {
-                    parse_meta(args, meta)?;
+    }
+
+    /// Verifies `#[reflect(skip_serializing)]` attributes are always last in the type definition,
+    /// but before `#[reflect(ignore)]` attributes.
+    fn check_skip_order(&self, args: &ReflectFieldAttr, errors: &mut Option<syn::Error>) {
+        if args.ignore == ReflectIgnoreBehavior::None {
+            if let Some(span) = self.last_skipped {
+                if self.is_variant {
+                    Self::combine_error(syn::Error::new(span, format!("fields marked with `#[reflect({IGNORE_SERIALIZATION_ATTR})]` must come last in variant definition (but before any fields marked `#[reflect({IGNORE_ALL_ATTR})]`)")), errors);
+                } else {
+                    Self::combine_error(syn::Error::new(span, format!("fields marked with `#[reflect({IGNORE_SERIALIZATION_ATTR})]` must come last in type definition (but before any fields marked `#[reflect({IGNORE_ALL_ATTR})]`)")), errors);
                 }
             }
-            Ok(())
+        }
+    }
+
+    /// Set or combine the given error into an optionally existing error.
+    fn combine_error(err: syn::Error, errors: &mut Option<syn::Error>) {
```

Comment:
Yeah I think I originally had it more specific to parsing fields so it made a _little_ more sense then. But now it's since been simplified down to the very basics. Maybe I'll move it to the `utility` module so it could be used elsewhere.